cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/ldap/LdapUserPersistence.java

```diff
@@ -160,33 +160,26 @@ public User addUserRequest(UserRequest userRequest, final Principal ownerHttpPri
         User user = null;
         try
         {
-            // create the group to be associated with this userRequest
+            // add the userRequest
             userDAO = new LdapUserDAO(conns);
+            user = userDAO.addUserRequest(userRequest);
+
+            // create the group to be associated with this userRequest
             LdapGroupDAO groupDAO = new LdapGroupDAO(conns, userDAO);
             LocalAuthority localAuthority = new LocalAuthority();
             URI gmsServiceURI = localAuthority.getServiceURI(Standards.GMS_GROUPS_01.toString());
             GroupURI groupID = new GroupURI(gmsServiceURI, userRequest.getUser().getHttpPrincipal().getName());
             Group group = new Group(groupID);
             User groupOwner = userDAO.getAugmentedUser(ownerHttpPrincipal,  false);
             ObjectUtil.setField(group, groupOwner, "owner");
-
-            // ensure that there is no existing group with the same user name
-            boolean groupExists = checkIfGroupExists(group, groupDAO);
-            if (groupExists) {
-                throw new GroupAlreadyExistsException("Group " + group.getID().getName() + " already exists ");
-            }
             
-            // group does not exist, add the userRequest
-            user = userDAO.addUserRequest(userRequest);
+            // add the user to the group
             group.getUserMembers().add(user);
             
             // add the group associated with the userRequest
             groupDAO.addUserAssociatedGroup(group, user.posixDetails.getGid());
         } catch (GroupAlreadyExistsException ex) {
-            // clean up
-            if (user != null) {
-                deleteUserRequest(user.getHttpPrincipal());
-            }
+            // no need to add the group, do nothing
```

Comment:
Are you saying that if the group already exists then that's okay?  A group exists for a reason so I don't think we hijack it for this new user...

Reply:
Added code to handle GroupAlreadyExistsException.

Reply:
It was really just a question about the logic... but I guess you agree that the existence of the group should be checked?